### PR TITLE
Stub of Dart 3 breaking change

### PR DESCRIPTION
**Pull request by mit-mit**
_Description of what this PR is changing or adding, and why:_
Initial draft of a new breaking change entry for Dart 3

_Issues fixed by this PR (if any):_ Fixes #ISSUE-NUMBER

## Presubmit checklist
- [ ] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [ ] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [ ] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Comment by MiraMarshall** [comment]
Thank you! @mit-mit I'm working on breaking changes for the next release and will take a look.

File: src/release/breaking-changes/dart-3.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+All code must be migrated to use full sound null safety. For details, see the
+[Dart null safety migration guide].
+
+## Removal of deprecated core library APIs
+
+Several currently `@deprecated` API members in the Dart core libraries are being
+removed in Dart 3.
+
+We believe these changes to have low impact on code migrated to use null safety.
+When we release the first Dart 3 alpha build -- in early 2032 -- you will be
```
Comment (MrBirb) [inline src/release/breaking-changes/dart-3.md:60]: nit: 2023 :-)

File: src/release/breaking-changes/dart-3.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
+---
+title: Dart 3 API removals & no unsound null safety support.
+description: Dart 3 will remove several deprecated APIs, and the ability to use unsound null safety.
```
Comment (sfshaza2) [inline src/release/breaking-changes/dart-3.md:3]:
```suggestion
description: Dart 3 removes several deprecated APIs, and the ability to use unsound null safety.
```

File: src/release/breaking-changes/dart-3.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+---
+
+## Summary
+
+The Dart language is preparing for a new major release, [Dart 3], in which
+there will be several breaking changes:
+
+* **No unsound null safety**: All code must run with sound null safety.
+* **Removal of deprecated core library APIs**: Several currently deprecated API
+  members in Dart core libraries are being removed.
```
Comment (sfshaza2) [inline src/release/breaking-changes/dart-3.md:13]:
```suggestion
The Dart language is preparing for a new major release, [Dart 3][], in which
there are several breaking changes:

* **No unsound null safety**: All code must run with _sound_ null safety.
* **Removal of deprecated core library APIs**: Several deprecated API
   members in Dart core libraries are being removed.
```

File: src/release/breaking-changes/dart-3.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+include this release in a Flutter SDK release around the same time.
+
+## No unsound null safety
+
+The Dart team is removing support for unsound null safety in Dart 3: 
+[Dart breaking change #49530]. Concretely this implies that Dart 3 will **not
+support**:
+
+1. Code using an SDK constraint where the min-constraint isn't at least 2.12
+   (e.g. `sdk: ">=2.7.0"` will be disallowed and won't resolve).
```
Comment (sfshaza2) [inline src/release/breaking-changes/dart-3.md:27]:
```suggestion
   (for example, `sdk: ">=2.7.0"` will be disallowed and won't resolve).
```

File: src/release/breaking-changes/dart-3.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+1. Libraries using [per-library language version markers] selecting versions
+   less than 2.12 (e.g. `// @dart=2.9` will be disallowed).
```
Comment (sfshaza2) [inline src/release/breaking-changes/dart-3.md:30]:
```suggestion
1. Libraries using [per-library language version markers][] selecting versions
   less than 2.12 (for example, `// @dart=2.9` will be disallowed).
```

File: src/release/breaking-changes/dart-3.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+environment:
+  sdk: '>=2.12.0 <3.0.0'
+```
+
+This allows developers to use Dart 3 sound null safety with packages that have been migrated to 2.12 null safety.
+
+### Migration guide
+
+All code must be migrated to use full sound null safety. For details, see the
+[Dart null safety migration guide].
```
Comment (sfshaza2) [inline src/release/breaking-changes/dart-3.md:52]:
```suggestion
All code must be migrated to use full sound null safety.
For details, see the [Dart null safety migration guide][].
```

File: src/release/breaking-changes/dart-3.md (hunk 1 above)
Comment (sfshaza2) [inline src/release/breaking-changes/dart-3.md:60]:
```suggestion
When we release the first Dart 3 alpha build -- in early 2023 -- you will be
```

**Review by sfshaza2 — APPROVED** [review]
Usually, we request that breaking change pages are written in past tense. But that won't work well in this case. But I've removed _some_ of the future tense.

**Comment by sfshaza2** [comment]
We'll have to update this page after Dart 3 launches, @MiraMarshall

**Comment by sfshaza2** [comment]
@mit-mit, we are freezing the website by end of day today. We will be closing this PR as part of that effort but it can be opened again after Flutter Forward.

**Comment by mit-mit** [comment]
OK. I'm not planning on doing anything further here -- this was opened to provide some input -- so can I take it that you'll make sure we get this landed at some point?

**Comment by sfshaza2** [comment]
Hmmm. I'll have to revisit this after Flutter Forward.

**Pull request closed without merging**